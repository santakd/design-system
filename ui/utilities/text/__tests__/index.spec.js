/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

describe('renders text utilities', () => {
  it('renders regular body text', () =>
    matchesMarkup(
      <div className="slds-text-body_regular">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders small body text', () =>
    matchesMarkup(
      <div className="slds-text-body_small">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders large heading text', () =>
    matchesMarkup(
      <div className="slds-text-heading_large">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders medium heading text', () =>
    matchesMarkup(
      <div className="slds-text-heading_medium">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders small heading text', () =>
    matchesMarkup(
      <div className="slds-text-heading_small">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders small title text', () =>
    matchesMarkup(
      <div className="slds-text-title">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders small title text in all-caps', () =>
    matchesMarkup(
      <div className="slds-text-title_caps">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders longform styles', () =>
    matchesMarkup(
      <div className="slds-text-longform">
        <p>The quick brown fox jumps over the lazy dog.</p>
        <p>The quick brown fox jumps over the lazy dog.</p>
        <h3 className="slds-text-heading_small">Heading</h3>
        <ul>
          <li>The quick brown fox jumps over the lazy dog.</li>
          <li>
            The quick brown fox jumps over the lazy dog.
            <ul>
              <li>The quick brown fox jumps over the lazy dog.</li>
              <li>
                The quick brown fox jumps over the lazy dog.
                <ul>
                  <li>The quick brown fox jumps over the lazy dog.</li>
                  <li>The quick brown fox jumps over the lazy dog.</li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
        <h3 className="slds-text-heading_small">Heading</h3>
        <ol>
          <li>The quick brown fox jumps over the lazy dog.</li>
          <li>
            The quick brown fox jumps over the lazy dog.
            <ol>
              <li>The quick brown fox jumps over the lazy dog.</li>
              <li>
                The quick brown fox jumps over the lazy dog.
                <ol>
                  <li>The quick brown fox jumps over the lazy dog.</li>
                  <li>The quick brown fox jumps over the lazy dog.</li>
                </ol>
              </li>
            </ol>
          </li>
        </ol>
      </div>
    ));

  it('renders left aligned text', () =>
    matchesMarkup(
      <div className="slds-text-align_left">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders center aligned text', () =>
    matchesMarkup(
      <div className="slds-text-align_center">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders right aligned text', () =>
    matchesMarkup(
      <div className="slds-text-align_right">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders default text color', () =>
    matchesMarkup(
      <div className="slds-text-color_default">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders weak text color', () =>
    matchesMarkup(
      <div className="slds-text-color_weak">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders error text color', () =>
    matchesMarkup(
      <div className="slds-text-color_error">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders destructive text color', () =>
    matchesMarkup(
      <div className="slds-text-color_destructive">
        The quick brown fox jumps over the lazy dog.
      </div>
    ));

  it('renders inversed text color', () =>
    matchesMarkup(
      <div style={{ padding: '0.5rem', background: '#16325c' }}>
        <div className="slds-text-color_inverse">
          The quick brown fox jumps over the lazy dog.
        </div>
      </div>
    ));

  it('renders inversed weak text color', () =>
    matchesMarkup(
      <div style={{ padding: '0.5rem', background: '#16325c' }}>
        <div className="slds-text-color_inverse-weak">
          The quick brown fox jumps over the lazy dog.
        </div>
      </div>
    ));

  it('renders monospace text font', () =>
    matchesMarkup(
      <div style={{ padding: '0.5rem', background: '#16325c' }}>
        <div className="slds-text-font_monospace">
          The quick brown fox jumps over the lazy dog.
        </div>
      </div>
    ));
});
