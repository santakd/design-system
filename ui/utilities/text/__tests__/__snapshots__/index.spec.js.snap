// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text utilities renders center aligned text 1`] = `
<div
  className="slds-text-align_center"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders default text color 1`] = `
<div
  className="slds-text-color_default"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders destructive text color 1`] = `
<div
  className="slds-text-color_destructive"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders error text color 1`] = `
<div
  className="slds-text-color_error"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders inversed text color 1`] = `
<div
  style={
    Object {
      "background": "#16325c",
      "padding": "0.5rem",
    }
  }
>
  <div
    className="slds-text-color_inverse"
  >
    The quick brown fox jumps over the lazy dog.
  </div>
</div>
`;

exports[`renders text utilities renders inversed weak text color 1`] = `
<div
  style={
    Object {
      "background": "#16325c",
      "padding": "0.5rem",
    }
  }
>
  <div
    className="slds-text-color_inverse-weak"
  >
    The quick brown fox jumps over the lazy dog.
  </div>
</div>
`;

exports[`renders text utilities renders large heading text 1`] = `
<div
  className="slds-text-heading_large"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders left aligned text 1`] = `
<div
  className="slds-text-align_left"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders longform styles 1`] = `
<div
  className="slds-text-longform"
>
  <p>
    The quick brown fox jumps over the lazy dog.
  </p>
  <p>
    The quick brown fox jumps over the lazy dog.
  </p>
  <h3
    className="slds-text-heading_small"
  >
    Heading
  </h3>
  <ul>
    <li>
      The quick brown fox jumps over the lazy dog.
    </li>
    <li>
      The quick brown fox jumps over the lazy dog.
      <ul>
        <li>
          The quick brown fox jumps over the lazy dog.
        </li>
        <li>
          The quick brown fox jumps over the lazy dog.
          <ul>
            <li>
              The quick brown fox jumps over the lazy dog.
            </li>
            <li>
              The quick brown fox jumps over the lazy dog.
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
  <h3
    className="slds-text-heading_small"
  >
    Heading
  </h3>
  <ol>
    <li>
      The quick brown fox jumps over the lazy dog.
    </li>
    <li>
      The quick brown fox jumps over the lazy dog.
      <ol>
        <li>
          The quick brown fox jumps over the lazy dog.
        </li>
        <li>
          The quick brown fox jumps over the lazy dog.
          <ol>
            <li>
              The quick brown fox jumps over the lazy dog.
            </li>
            <li>
              The quick brown fox jumps over the lazy dog.
            </li>
          </ol>
        </li>
      </ol>
    </li>
  </ol>
</div>
`;

exports[`renders text utilities renders medium heading text 1`] = `
<div
  className="slds-text-heading_medium"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders monospace text font 1`] = `
<div
  style={
    Object {
      "background": "#16325c",
      "padding": "0.5rem",
    }
  }
>
  <div
    className="slds-text-font_monospace"
  >
    The quick brown fox jumps over the lazy dog.
  </div>
</div>
`;

exports[`renders text utilities renders regular body text 1`] = `
<div
  className="slds-text-body_regular"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders right aligned text 1`] = `
<div
  className="slds-text-align_right"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders small body text 1`] = `
<div
  className="slds-text-body_small"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders small heading text 1`] = `
<div
  className="slds-text-heading_small"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders small title text 1`] = `
<div
  className="slds-text-title"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders small title text in all-caps 1`] = `
<div
  className="slds-text-title_caps"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;

exports[`renders text utilities renders weak text color 1`] = `
<div
  className="slds-text-color_weak"
>
  The quick brown fox jumps over the lazy dog.
</div>
`;
